source/mir/glas/dot.d

```diff
@@ -0,0 +1,145 @@
+/++
+$(H2 Dot Product)
+
+$(SCRIPT inhibitQuickIndea = 1;)
+
+This is a submodule of $(LINK2 mir_glas.html, mir.glas).
+
+License: $(LINK2 http://boost.org/LICENSE_1_0.tat, Boost License 1.0).
+
+Authors: Ilya Yaroshenko
+
+Macros:
+SUBMODULE = $(LINK2 mir_glas_$1.html, mir.glas.$1)
+SUBREF = $(LINK2 mir_glas_$1.html#.$2, $(TT $2))$(NBSP)
++/
+module mir.glas.dot;
+
+import std.traits;
+import std.range.primitives;
+import mir.ndslice.slice;
+import mir.internal.utility;
+
+public import mir.glas.common;
+
+@fastmath:
+
+/++
+Computes a dot product (inner product) of two vectors.
+
+Params:
+    a = vector
+    b = vector
+    S = summation type, optional template argument
+    type = conjugating type, e.g. `dot!ConjA(a, b)` conjugate first vector. Optinal template argument.
+
+Returns: scalar `aᵀ × b`
+BLAS: SDOT, SDSDOT, DDOT, DSDOT, CDOTU, CDOTC, ZDOTU, ZDOTC
+
+See_also: $(SUBREF common, Conjugation)
++/
+pure nothrow @nogc
```

Comment:
I thought explicit annotation isn't recommended as it won't recognize/annotate `@safe`?